### PR TITLE
axbacktrace: emit raw backtrace blocks for host-side symbolize

### PR DESCRIPTION
**Pull request by Jiaxin2006**
Tracking issue: #146

- Summary: Print stable BACKTRACE_BEGIN/BT/BACKTRACE_END blocks on panic path so host-side tools can symbolize.
- Rationale: Avoid target-side DWARF parsing risk on panic/oops paths.
- Tests:
  - cargo test -p axbacktrace -- --test-threads=1
  - cargo check -p axbacktrace --target aarch64-unknown-none-softfloat
  - cargo check -p ax-runtime --target aarch64-unknown-none-softfloat

### CODE REVIEW

File: test-suit/arceos/rust/backtrace-raw-basic/qemu-x86_64.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "128M",
+    "-smp",
+    "4",
+    "-nographic",
+    "-serial",
+    "mon:stdio",
+]
+uefi = false
+to_bin = false
+success_regex = [
```
Comment (ZR233) [inline test-suit/arceos/rust/backtrace-raw-basic/qemu-x86_64.toml:16]: 此 success_regex 要求匹配 `^BT\s+0\s+ip=0x[0-9a-fA-F]+\s+fp=0x[0-9a-fA-F]+`，但本地运行时实际输出为 `BT_ERROR invalid_fp`，没有 `BT 0` 行。success_regex 并未真正验证到 backtrace 帧，需要修正 regex 或确保测试在 frame pointers 启用的配置下运行。

File: test-suit/arceos/rust/backtrace-raw-basic/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+app! {
+
+#[cfg(feature = "ax-std")]
+extern crate ax_std as std;
+
+use axbacktrace::Backtrace;
+use std::println;
+
+fn c() {
```
Comment (ZR233) [inline test-suit/arceos/rust/backtrace-raw-basic/src/main.rs:24]: 测试名为 `backtrace-panic` 但不触发 panic，直接调用 `Backtrace::report("raw")`。同时 `lang_items.rs` 中 panic 处理器已改为 `Backtrace::report("panic")`，但 panic 路径无任何 QEMU 测试覆盖。建议恢复 panic 行为测试或重命名此测试。

**Review by ZR233 — CHANGES_REQUESTED** [review]
## 审查总结

**注意：此 PR 的 5 个 commit 与 PR #646 的前 5 个 commit 完全相同（相同 SHA）。** PR #646 在此基础上额外增加了第 6 个 commit（backtrace-raw-normal/badfp 测试用例）。两个 PR 的核心变更和问题完全一致。

这个 PR 为 axbacktrace 添加了机器可读的原始 backtrace 报告格式，实现逻辑合理：
- `BacktraceReport` 输出结构化的 `BACKTRACE_BEGIN/END` 格式，包含 FP 链校验
- `read_current_fp()` 从内联汇编重构为独立函数
- 新增 4 个单元测试和 1 个集成测试

### 阻塞问题

1. **`backtrace-panic` 的 success_regex 与实际输出不匹配**

   `qemu-*.toml` 中 `success_regex` 要求匹配 `^BT\s+0\s+ip=0x...` 但本地运行时实际输出为 `BT_ERROR invalid_fp`，没有 `BT 0` 行。测试虽然通过了，但 success_regex 并未真正验证到 backtrace 帧。

   本地运行结果：
   ```
   BACKTRACE_BEGIN kind=raw arch=x86_64 alloc=false dwarf=false
   BT_ERROR invalid_fp
   BACKTRACE_END
   ```

2. **`backtrace-panic` 测试名不反映实际行为，且 panic 路径无测试覆盖**

   该测试直接调用 `Backtrace::report("raw")` 而不是触发 panic。同时 `lang_items.rs` 中的 panic 处理器已被改为使用 `Backtrace::report("panic")`，但该 panic 路径没有任何 QEMU 测试覆盖。`fail_regex = ["(?i)\\bpanic(?:ked)?\\b"]` 也会阻止测试 panic 行为。

### 验证结果

| 命令 | 结果 |
|------|------|
| `cargo fmt --check` | ✅ 通过 |
| `cargo clippy -p axbacktrace --all-features` | ✅ 通过 |
| `cargo test -p axbacktrace` | ✅ 5/5 通过 |
| `cargo check -p axbacktrace --target aarch64-unknown-none-softfloat` | ✅ 通过 |
| `cargo xtask arceos test qemu --arch x86_64 -c backtrace-panic` | ✅ 通过（但 success_regex 不完全匹配） |

### CI 状态

- `Test starry x86_64 qemu` 失败：`card0.rs` 中 `DeviceOps::mmap` 方法签名不匹配（2 vs 3 参数）。此文件未被本 PR 修改，是 base 分支的预存问题，与本 PR 无关。

**Comment by Jiaxin2006** [comment]
已根据 CHANGES_REQUESTED 修正：
- 将名不副实的 backtrace-panic 重命名为 backtrace-raw-basic（仅做 raw 冒烟，避免与 panic-path 语义冲突）
- 修正 QEMU 断言：success_regex 改为单条多行正则，避免多条 success_regex 的 OR 假绿，确保必须包含 BACKTRACE_BEGIN(kind=raw)+BT 0+BACKTRACE_END+test pass
- build-*.toml 改用 DWARF=y/BACKTRACE=y 由构建系统注入 rustflags（避免覆盖链接 rustflags）

本地 x86_64 QEMU 用例已验证通过。
由于当前 runner 全局 fail pattern 会匹配 panic/panicked（含 kind=panic），所以无法把“真正触发 panic 走 lang_items panic handler”做成成功用例；因此先通过重命名明确该用例职责为 raw smoke。

**Comment by Jiaxin2006** [comment]
Closing in favor of #619 (raw backtrace report + arceos backtrace test) and #646 (qemu raw backtrace edge cases + host-side symbolize). All changes in #640 are included in #646, and #619/#646 are tracked under issue #146.
